vm/nova/vm.go

```diff
@@ -689,6 +668,111 @@ func accountDestructionValid(vmParams *vm.Params, input *vm.ChainOutputWithIDs)
 	return nil
 }
 
+// For output AnchorOutput(s) with non-zeroed AnchorID, there must be a corresponding input AnchorOutput where either its
+// AnchorID is zeroed and StateIndex is zero or an input AnchorOutput with the same AnchorID.
+//
+// On anchor state transitions: The StateIndex must be incremented by 1 and Only Amount, StateIndex and StateMetadata can be mutated.
+//
+// On anchor governance transition: Only StateController (must be mutated), GovernanceController and the MetadataBlock can be mutated.
```

Comment:
Is it true that the state controller must be mutated? I don't see that enforced.